packages/nuxt/src/app/composables/chunk.ts

```diff
@@ -36,10 +42,12 @@ export function reloadNuxtApp (options: ReloadNuxtAppOptions = {}) {
       sessionStorage.setItem('nuxt:reload', JSON.stringify({ path, expires: Date.now() + (options.ttl ?? 10000) }))
     } catch {}
 
-    try {
-      // TODO: handle serializing/deserializing complex states as JSON: https://github.com/nuxt/nuxt/pull/19205
-      sessionStorage.setItem('nuxt:reload:state', JSON.stringify({ state: useNuxtApp().payload.state }))
-    } catch {}
+    if (options.persistState) {
+      try {
+        // TODO: handle serializing/deserializing complex states as JSON: https://github.com/nuxt/nuxt/pull/19205
+        sessionStorage.setItem('nuxt:reload:state', JSON.stringify({ state: useNuxtApp().payload.state }))
+      } catch {}
+    }
```

Comment:
(I think it might be safe to remove `nuxt:reload:state` otherwise to ensure there is no stalled data)

Reply:
If `experimental.restoreState` is set, then we will do that on reload. Otherwise the user is handling it manually and will need to clean it up themselves.